torch/_dynamo/utils.py

```diff
@@ -2651,3 +2651,69 @@ def wrapper(*args):
         return compiled_fn(*pytree.arg_tree_leaves(*args))
 
     return wrapper
+
+# This is used to proxy call to subclass constructors:
+# - Subclass()
+# - Subclass.__new__()
+# - Subclass.__init__()
+def proxy_fn_with_non_fx_args(fn, tx, args, kwargs):
```

Comment:
@wanchaol does this subsume the thing you did for DTensor?

Reply:
yep, this looks bascially the lamaba closure function I did for DTensor